app/models/ability.rb

```diff
@@ -0,0 +1,38 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if user.is?(:admin)
+      can :manage, :all
+    else
+      can :read, :all
+      can :destroy, Post, author: user
+      can :destroy, Comment, author: user
```

Comment:
- [ ] Please make sure to use `:manage` instead `:destroy` for posts and comments for their authors. `:manage` will allow the authors to perform all actions view, update and delete.

Reply:
- Thank you for the suggestion, I have implemented it. 😄